src/funnels/automation/StableDepositor.sol

```diff
@@ -61,15 +61,25 @@ interface DepositorLike {
 contract StableDepositor {
     mapping (address => uint256) public wards;                           // Admins
     mapping (address => uint256) public buds;                            // Whitelisted keepers
-    mapping (address => mapping (address => PairConfig)) public configs; // Configuration for keepers
+    mapping (address => mapping (address => mapping (uint24 => mapping (int24 => mapping (int24 => PairConfig))))) public configs; // Configuration for keepers
 
     DepositorLike public immutable depositor; // Depositor for this StableDepositor
 
+    struct PairConfig {
+        uint32 num;  // The remaining number of times that a (gem0, gem1) operation can be performed by keepers
```

Comment:
One open question is whether it could make sense to make this a signed integer representing the amount of times `deposit()` can be called when positive, or the amount of time `withdraw()` can be called when negative. Right now we update the same `num` for both deposits and withdrawals but maybe we should be more consistent with how we treat the `num` in the StableSwapper where we have a different `num` for each swap direction (and generally expect one of the two swap directions to have its `num == 0`).

Reply:
Yeah makes sense.
Think of a keeper script that just simulates a tx and calls it on chain in case it doesn't revert.
How would it know what function to call (deposit or withdraw) without this?

I personally don'y like much using signed integers, so would prefer another variable to indicate the direction (especially as it seem we have some place in the packed struct).
Can check it in the same require as we check `cfg.num`.

Reply:
The good thing about the signed integer is that easily enforces that they are mutually exclusive which I'd think it makes more sense. But if you insist in two variables I can give a try, worst case we can split in two uint16 which allows much less number but it is still more than enough for this use case.
Anyway I think I'd prefer to manage this change in a specific PR once this one gets merged.

Reply:
Ended up making this sub PR to this PR: https://github.com/makerdao/dss-allocator/pull/29